Support of Custom Code

It seems that custom code is only partly supported in grtfmi.tlc. 

We tried to declare a define. It seems  that option is not correctly passed to compiler (Visual Studio or MinGW).
Way to reproduce:

1. Declare header code (through GUI):
```
#ifdef TOTO
# error "TOTO is well defined"
#else
# error "TOTO is not defined"
#endif
```

2. Declare 'TOTO' in define list (through GUI).

Compilation with grtfmi.tlc leads to 
```
...\test_define.h:45:3: error: #error "TOTO is not defined"
 # error "TOTO is not defined"
```

 Compilation with grt.tlc leads to expected "TOTO is well defined".

I attached a SLX to reproduce de issue.

[test_define.zip](https://github.com/CATIA-Systems/FMIKit-Simulink/files/10863067/test_define.zip)

--- a/grtfmi/CMakeLists.txt
+++ b/grtfmi/CMakeLists.txt
@@ -212,6 +212,7 @@ target_compile_definitions(${MODEL_NAME} PUBLIC
   USE_RTMODEL
   NO_FMI_FUNCTION_PREFIX
   GRTFMI
+  ${CUSTOM_DEFINE}
 )
 
 target_link_libraries(${MODEL_NAME} ${CUSTOM_LIBRARY})

--- a/grtfmi/grtfmi_make_rtw_hook.m
+++ b/grtfmi/grtfmi_make_rtw_hook.m
@@ -83,6 +83,7 @@ disp('### Running CMake generator')
 custom_include = cmake_list(custom_include);
 custom_source  = cmake_list(custom_source);
 custom_library = cmake_list(custom_library);
+custom_define  = cmake_list(regexp(get_param(modelName, 'CustomDefine'), '\s+', 'split'));
 
 % check for Simscape blocks
 if isempty(find_system(modelName, 'BlockType', 'SimscapeBlock'))
@@ -106,6 +107,7 @@ fprintf(fid, 'MATLAB_ROOT:STRING=%s\n', strrep(matlabroot, '\', '/'));
 fprintf(fid, 'CUSTOM_INCLUDE:STRING=%s\n', custom_include);
 fprintf(fid, 'CUSTOM_SOURCE:STRING=%s\n', custom_source);
 fprintf(fid, 'CUSTOM_LIBRARY:STRING=%s\n', custom_library);
+fprintf(fid, 'CUSTOM_DEFINE:STRING=%s\n', custom_define);
 fprintf(fid, 'SOURCE_CODE_FMU:BOOL=%s\n', upper(source_code_fmu));
 fprintf(fid, 'SIMSCAPE:BOOL=%s\n', upper(simscape_blocks));
 fprintf(fid, 'FMI_VERSION:STRING=%s\n', fmi_version);

